wgsl/index.bs

```diff
@@ -896,33 +900,143 @@ A <dfn noexport>type rule</dfn> has two parts:
     * How the expression is used in a statement.
     * Optionally, other static context.
 
-Each distinct type parameterization for a type rule is called an <dfn noexport>overload</dfn>.
-For example, [[#arithmetic-expr|unary negation]] (an expression of the form `-`|e|)
-has eight overloads, because its type rules are parameterized by a type |T| that can be any of:
-* [=i32=]
-* vec2&lt;i32&gt;
-* vec3&lt;i32&gt;
-* vec4&lt;i32&gt;
-* [=f32=]
-* vec2&lt;f32&gt;
-* vec3&lt;f32&gt;
-* vec4&lt;f32&gt;
+Type rules may have <dfn noexport>type parameters</dfn> in their preconditions and conclusions.
+When a type rule's conclusion or preconditions contain type parameters,
+we say it is <dfn noexport>parameterized</dfn>.
+When they do not, we say the rule is <dfn noexport>concrete</dfn>.
+We can make a concrete type rule from a parameterized one
+by substituting a type for each of its type parameters,
+using the same type for all occurrences of a given parameter in the rule.
+An assignment of types to a rule's type parameters
+is called a <dfn noexport>substitution</dfn>.
+
+For example, the two type rules for [[#arithmetic-expr|unary negation]] (an expression of the form `-`|e|)
+are as follows:
+
+<table class='data'>
+  <thead>
+    <tr><th>Precondition<th>Conclusion
+  </thead>
+  <tr algorithm="example integer negation"><td>|e|`:` |T|<br>
+  |T| is [SIGNEDINTEGRAL]
+  <td>`-`|e|`:` |T|
+
+  <tr algorithm="example floating point negation"><td>|e|`:` |T|<br>|T| is [FLOATING]
+  <td>`-`|e|`:` |T|
+</table>
+
+These are both parameterized rules, because they contain the type parameter `T`.
+In the first type rule, `T` can represent any one of four types
+[=i32=], `vec2<i32>`, `vec3<i32>`, or `vec4<i32>`.
+In the second, it can represent any one of the four types
+[=f32=], `vec2<f32>`, `vec3<f32>`, or `vec4<f32>`.
+Applying the substitution that maps `T` to `vec3<i32>`
+produces the concrete type rule:
+
+<table class='data'>
+  <thead>
+    <tr><th>Precondition<th>Conclusion
+  </thead>
+  <tr algorithm="example2 integer negation"><td>|e|`: vec3<i32>`
+  <td>`-`|e|`: vec3<i32>`
+</table>
+
+Each concrete rule we can produce from a parameterized rule
```

Comment:
As written, if there are multiple concrete rules for the same construct, but they are not written in the form of a single parameterized rule it does not count as an overload. Is it intended? Or is it just that currently that case does not appear anywhere in the spec?